README.md

```diff
@@ -25,3 +25,20 @@ The app deployed as part of this solution does the following:
 
 The agent crashes whenever it encounters an unrecoverable error with the metadata APIs.
 This agent is not production hardened yet and so use it with caution.
+
+## Graceful terminations for regular pods (Non-system pods)
+
+The pods that are not in the kube-system are called **regular pods** in this agent.
+By default, regular pods are deleted immediately before deleting system pods.
+If you want to delete regular pods gracefully, please add `--system-pod-grace-period=n` in arguments according to the following rules:
+
+- If targeted VM is Preemptible VM, specify `n` with a value from `0s` to `14s`.
+- If targeted VM is regular VM, specify `n` with a value from `0s` to the value of `(--regular-vm-timeout / 2) - 1`.
+
+If you follow the rules above, `VM timeout - system-grace-pod-period` will be given as a grace period for deleting regular pods.
+
+Also, `the timeout value of VM (e.g. preemptible=30s) / 2` cannot be used as a maximum value for the `--system-pod-grace-period`.
+This is because, the timing at which these delete events are triggered by a subscriber to GCE's Metadata API is different from the timing of the actual delete process.
+In other words, with proper consideration of the elapsed time, it's not possible to take advantage of all the duration remaining from the time a VM is known to be scheduled to delete until the VM is deleted actually.
+
+In addition, if the actual delete process fails, it will retry internally based on exponential backoff. In that case, the grace period is set considering the elapsed time, but it may shorten the actual grace period.
```

Comment:
where does the exponential backoff happen? My understanding is that the only deletion is [here](https://github.com/GoogleCloudPlatform/k8s-node-termination-handler/blob/master/termination/eviction.go#L95).

Reply:
Here: https://github.com/GoogleCloudPlatform/k8s-node-termination-handler/blob/d56ae034967302518f8613d7333de5610965fd23/termination/handler.go#L99

Since the `processNodeState` is covered by the function and the deletion code is performed inside the lifecycle of `processNodeState`, I think there is a possibility of shortening the grace period by the retry methodology.

Reply:
Got it, thanks for the explanation.